### PR TITLE
use docker healthcheck status for readiness

### PR DESCRIPTION
**Pull request by tannerjfco**
## The Problem:
Our existing readiness check tests for a 200 status on the healthcheck endpoint of our web container. This works great for the web container, but it is not useful for determining readiness of any other container.

## The Fix:
Docker introduced `HEALTHCHECK` in 1.12. We should leverage it, so that ddev can determine the readiness of any container, so long as a `HEALTHCHECK CMD` is defined for the container.

This will help to be able to satisfy the requirement that `import-db` should not run unless the site is provisioned. Checking the db container seems like the most relevant check in this case. This will also help in writing a test case for import, as I need to wait for the db container to be ready.

## The Test:
On a test site that you have run `ddev config`, update the tags for both web and db images to the `docker-health` tag and run `ddev start`. The readiness check will use the new docker health check status. If you are using the `docker-health` tag, the site should report ready as expected.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ DdevVersion ?= $(VERSION)
 WebImg ?= drud/nginx-php-fpm7-local
 WebTag ?= v0.3.4
 DBImg ?= drud/mysql-docker-local-57
-DBTag ?= v0.2.0
+DBTag ?= v0.2.1
```
Comment (rfay) [inline Makefile]: DBTag [v0.3.0](https://github.com/drud/mysql-docker-local/releases/tag/v0.3.0) was already tagged by one @tannerjfco 3 days ago. Ya gotta keep up! Probably just good to rebase this PR.

File: pkg/plugins/platform/local.go [hunk 2]
```diff
@@ -427,3 +425,24 @@ func (l *LocalApp) AddHostsEntry() error {
 	err = system.RunCommandPipe("sudo", hostnameArgs)
 	return err
 }
+
+// ContainerWait provides a wait loop to check for container in "healthy" status.
+func (l *LocalApp) ContainerWait(timeout time.Duration, containerType string) error {
```
Comment (rfay) [inline pkg/plugins/platform/local.go]: Don't we want this in a more general location? Seems like a candidate for pkg/util/docker.go (in https://github.com/drud/ddev/pull/146

File: pkg/plugins/platform/utils.go [hunk 3]
```diff
@@ -394,3 +396,18 @@ func FindContainerByLabels(labels map[string]string) (docker.APIContainers, erro
 
 	return docker.APIContainers{}, fmt.Errorf("could not find containers which matched search criteria: %+v", labels)
 }
+
+// GetContainerHealth retrieves the status of a given container. The status string returned
+// by docker contains uptime and the health status in parenths. This function will filter the uptime and
+// return only the health status.
+func GetContainerHealth(container docker.APIContainers) string {
```
Comment (rfay) [inline pkg/plugins/platform/utils.go]: I think this should go in a more general location as well.

File: pkg/version/version.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ var WebTag = "v0.3.4" // Note that this is overridden by make
 var DBImg = "drud/mysql-docker-local-57" // Note that this is overridden by make
 
 // DBTag defines the default db image tag for drud dev
-var DBTag = "v0.2.0" // Note that this is overridden by make
+var DBTag = "v0.2.1" // Note that this is overridden by make
```
Comment (rfay) [inline pkg/version/version.go]: Whatever current is is what it should be. Not sure when 0.3.0 comes in.

**Comment by rfay** [comment]
This still has the "Needs tests" label - is it ready to go? Since the tests explicitly use it, maybe that's it?

**Comment by tannerjfco** [comment]
I ended up reworking and making quite a few changes after approval, so this could probably use another review. I have also addressed a couple outliers in this PR:

- Moved util sub pkgs back to util main pkg
- #123 introduced a test for ImportSQLDump, but it was disabled as it needed this work in order to succeed. It has been updated as a working test.
- I also changed app.Wait() for better flexibility to use for other containers aside from web.

File: pkg/util/dockerutils_test.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestGetContainerHealth tests the function for processing container readiness.
+func TestGetContainerHealth(t *testing.T) {
+	assert := assert.New(t)
+	container := docker.APIContainers{
+		Status: "Up 24 seconds (health: starting)",
```
Comment (beeradb) [inline pkg/util/dockerutils_test.go]: very nice.

File: pkg/util/dockerutils.go [hunk 6]
```diff
@@ -175,3 +178,39 @@ func NetExists(client *docker.Client, name string) bool {
...
+func GetContainerHealth(container docker.APIContainers) string {
+	status := container.Status
+	re := regexp.MustCompile(`\(([^\)]+)\)`)
+	match := re.FindString(status)
+	match = strings.Trim(match, "()")
+	pre := "health: "
+	if strings.HasPrefix(match, pre) {
+		match = strings.TrimPrefix(match, pre)
+	}
+	return match
```
Comment (beeradb) [inline pkg/util/dockerutils.go]: it feels a bit strange to me that there's no potential for error in any of this.

**Pull request merged**